chain/default_syncer.go

```diff
@@ -334,6 +341,7 @@ func (syncer *DefaultSyncer) logReorg(ctx context.Context, curHead, newHead type
 	if reorg {
 		dropped, added, err := ReorgDiff(curHead, newHead, commonAncestor)
 		if err == nil {
+			reorgCnt.Inc(ctx, 1)
```

Comment:
I think we should move this up a level. It looks like the reorg still occurs, and this check is only for calculating the diff.